openedx/features/partners/views.py

```diff
@@ -0,0 +1,11 @@
+from django.shortcuts import get_object_or_404
+from g2a.views import g2a_dashboard
+from .models import Partner
+
+
+def dashboard(request, slug):
+    partner = get_object_or_404(Partner, slug=slug)
+    # TODO: we need to make it generic for all partners
+    if partner.slug == 'give2asia':
+        return g2a_dashboard(request, partner.slug)
+
```

Comment:
return 404 response here as well to handle if slug is added in admin but not implemented in code.